src/OpenTK/MouseCursorFrame.cs

```diff
@@ -0,0 +1,135 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+#endregion
+
+using System;
+
+namespace OpenTK
+{
+    /// <summary>
+    /// Represents a frame for a predefined or custom mouse cursor.
+    /// </summary>
+    public sealed class MouseCursorFrame : WindowIcon
+    {
+        static readonly MouseCursorFrame default_cursor = new MouseCursorFrame();
+        static readonly MouseCursorFrame empty_cursor = new MouseCursorFrame(
+            0, 0, 16, 16, new byte[16 * 16 * 4]);
+
+        int x;
+        int y;
+
+        MouseCursorFrame()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new <see cref="MouseCursorFrame"/> instance from a
+        /// contiguous array of BGRA pixels.
+        /// Each pixel is composed of 4 bytes, representing B, G, R and A values,
+        /// respectively. For correct antialiasing of translucent cursors,
+        /// the B, G and R components should be premultiplied with the A component:
+        /// <code>
+        /// B = (byte)((B * A) / 255)
+        /// G = (byte)((G * A) / 255)
+        /// R = (byte)((R * A) / 255)
+        /// </code>
+        /// </summary>
+        /// <param name="hotx">The x-coordinate of the cursor hotspot, in the range [0, width]</param>
+        /// <param name="hoty">The y-coordinate of the cursor hotspot, in the range [0, height]</param>
+        /// <param name="width">The width of the cursor data, in pixels.</param>
+        /// <param name="height">The height of the cursor data, in pixels.</param>
+        /// <param name="data">
+        /// A byte array representing the cursor image,
+        /// laid out as a contiguous array of BGRA pixels.
+        /// </param>
+        public MouseCursorFrame(int hotx, int hoty, int width, int height, byte[] data)
+            : base(width, height, data)
+        {
+            if (hotx < 0 || hotx >= Width || hoty < 0 || hoty >= Height)
+                throw new ArgumentOutOfRangeException();
```

Comment:
Can we include the constraints in the documentation for this constructor?

Optional, but nice to have would be clear explanations as to _which_ argument is out of range, (and what it's value was) would be useful.